{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e84e7bdd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<gzip _io.BufferedReader name='SNV/02747363-f04a-4ba6-a079-fe4f87853788/TCGA.UCS.mutect.02747363-f04a-4ba6-a079-fe4f87853788.DR-10.0.somatic.maf.gz' 0x1fd8d78b4f0>\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import time\n",
    "import pandas as pd\n",
    "from sklearn import preprocessing\n",
    "from sklearn.impute import SimpleImputer\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split\n",
    "import gzip\n",
    "\n",
    "df = pd.read_table(\"SNV/MANIFEST.txt\")\n",
    "filelist = []\n",
    "header = []\n",
    "for file_name in df[0:33][\"filename\"]:\n",
    "    filelist.append(file_name)\n",
    "\n",
    "#for m in filelist:\n",
    " #   print(m)\n",
    "with gzip.open(\"SNV/\"+filelist[0],\"rb\") as f:\n",
    "    print(f)\n",
    "    i = 0\n",
    "    while i <=4:\n",
    "        header.append(f.readline())\n",
    "        i += 1\n",
    "    mf = pd.read_csv(f, delimiter = \"\\t\")\n",
    "    table = mf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "681d7c36",
   "metadata": {},
   "outputs": [],
   "source": [
    "#header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ec6614be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "512307"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table.isna().sum().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f114d5c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Data columns (total 120 columns):\n",
      " #    Column                         Non-Null Count  Dtype  \n",
      "---   ------                         --------------  -----  \n",
      " 0    Hugo_Symbol                    10449 non-null  object \n",
      " 1    Entrez_Gene_Id                 10449 non-null  int64  \n",
      " 2    Center                         10449 non-null  object \n",
      " 3    NCBI_Build                     10449 non-null  object \n",
      " 4    Chromosome                     10449 non-null  object \n",
      " 5    Start_Position                 10449 non-null  int64  \n",
      " 6    End_Position                   10449 non-null  int64  \n",
      " 7    Strand                         10449 non-null  object \n",
      " 8    Variant_Classification         10449 non-null  object \n",
      " 9    Variant_Type                   10449 non-null  object \n",
      " 10   Reference_Allele               10449 non-null  object \n",
      " 11   Tumor_Seq_Allele1              10449 non-null  object \n",
      " 12   Tumor_Seq_Allele2              10449 non-null  object \n",
      " 13   dbSNP_RS                       9507 non-null   object \n",
      " 14   dbSNP_Val_Status               1777 non-null   object \n",
      " 15   Tumor_Sample_Barcode           10449 non-null  object \n",
      " 16   Matched_Norm_Sample_Barcode    10449 non-null  object \n",
      " 17   Match_Norm_Seq_Allele1         0 non-null      float64\n",
      " 18   Match_Norm_Seq_Allele2         0 non-null      float64\n",
      " 19   Tumor_Validation_Allele1       2628 non-null   object \n",
      " 20   Tumor_Validation_Allele2       2628 non-null   object \n",
      " 21   Match_Norm_Validation_Allele1  0 non-null      float64\n",
      " 22   Match_Norm_Validation_Allele2  0 non-null      float64\n",
      " 23   Verification_Status            0 non-null      float64\n",
      " 24   Validation_Status              0 non-null      float64\n",
      " 25   Mutation_Status                10449 non-null  object \n",
      " 26   Sequencing_Phase               0 non-null      float64\n",
      " 27   Sequence_Source                0 non-null      float64\n",
      " 28   Validation_Method              2628 non-null   object \n",
      " 29   Score                          0 non-null      float64\n",
      " 30   BAM_File                       0 non-null      float64\n",
      " 31   Sequencer                      10449 non-null  object \n",
      " 32   Tumor_Sample_UUID              10449 non-null  object \n",
      " 33   Matched_Norm_Sample_UUID       10449 non-null  object \n",
      " 34   HGVSc                          10364 non-null  object \n",
      " 35   HGVSp                          9002 non-null   object \n",
      " 36   HGVSp_Short                    9134 non-null   object \n",
      " 37   Transcript_ID                  10449 non-null  object \n",
      " 38   Exon_Number                    9828 non-null   object \n",
      " 39   t_depth                        10449 non-null  int64  \n",
      " 40   t_ref_count                    10449 non-null  int64  \n",
      " 41   t_alt_count                    10449 non-null  int64  \n",
      " 42   n_depth                        10449 non-null  int64  \n",
      " 43   n_ref_count                    0 non-null      float64\n",
      " 44   n_alt_count                    0 non-null      float64\n",
      " 45   all_effects                    10449 non-null  object \n",
      " 46   Allele                         10449 non-null  object \n",
      " 47   Gene                           10449 non-null  object \n",
      " 48   Feature                        10449 non-null  object \n",
      " 49   Feature_type                   10449 non-null  object \n",
      " 50   One_Consequence                10449 non-null  object \n",
      " 51   Consequence                    10449 non-null  object \n",
      " 52   cDNA_position                  10449 non-null  object \n",
      " 53   CDS_position                   10247 non-null  object \n",
      " 54   Protein_position               10247 non-null  object \n",
      " 55   Amino_acids                    9003 non-null   object \n",
      " 56   Codons                         9003 non-null   object \n",
      " 57   Existing_variation             2409 non-null   object \n",
      " 58   ALLELE_NUM                     10449 non-null  int64  \n",
      " 59   DISTANCE                       85 non-null     float64\n",
      " 60   TRANSCRIPT_STRAND              10449 non-null  int64  \n",
      " 61   SYMBOL                         10449 non-null  object \n",
      " 62   SYMBOL_SOURCE                  10449 non-null  object \n",
      " 63   HGNC_ID                        10369 non-null  object \n",
      " 64   BIOTYPE                        10449 non-null  object \n",
      " 65   CANONICAL                      8263 non-null   object \n",
      " 66   CCDS                           9760 non-null   object \n",
      " 67   ENSP                           10247 non-null  object \n",
      " 68   SWISSPROT                      9934 non-null   object \n",
      " 69   TREMBL                         1748 non-null   object \n",
      " 70   UNIPARC                        9135 non-null   object \n",
      " 71   RefSeq                         9372 non-null   object \n",
      " 72   SIFT                           5594 non-null   object \n",
      " 73   PolyPhen                       5988 non-null   object \n",
      " 74   EXON                           9828 non-null   object \n",
      " 75   INTRON                         541 non-null    object \n",
      " 76   DOMAINS                        6543 non-null   object \n",
      " 77   GMAF                           442 non-null    float64\n",
      " 78   AFR_MAF                        479 non-null    float64\n",
      " 79   AMR_MAF                        479 non-null    float64\n",
      " 80   ASN_MAF                        0 non-null      float64\n",
      " 81   EAS_MAF                        479 non-null    float64\n",
      " 82   EUR_MAF                        479 non-null    float64\n",
      " 83   SAS_MAF                        479 non-null    float64\n",
      " 84   AA_MAF                         601 non-null    float64\n",
      " 85   EA_MAF                         601 non-null    float64\n",
      " 86   CLIN_SIG                       100 non-null    object \n",
      " 87   SOMATIC                        10 non-null     object \n",
      " 88   PUBMED                         33 non-null     object \n",
      " 89   MOTIF_NAME                     0 non-null      float64\n",
      " 90   MOTIF_POS                      0 non-null      float64\n",
      " 91   HIGH_INF_POS                   0 non-null      float64\n",
      " 92   MOTIF_SCORE_CHANGE             0 non-null      float64\n",
      " 93   IMPACT                         10449 non-null  object \n",
      " 94   PICK                           7916 non-null   float64\n",
      " 95   VARIANT_CLASS                  10449 non-null  object \n",
      " 96   TSL                            9823 non-null   float64\n",
      " 97   HGVS_OFFSET                    225 non-null    float64\n",
      " 98   PHENO                          102 non-null    object \n",
      " 99   MINIMISED                      10449 non-null  int64  \n",
      " 100  ExAC_AF                        2137 non-null   float64\n",
      " 101  ExAC_AF_Adj                    2137 non-null   float64\n",
      " 102  ExAC_AF_AFR                    2137 non-null   float64\n",
      " 103  ExAC_AF_AMR                    2137 non-null   float64\n",
      " 104  ExAC_AF_EAS                    2137 non-null   float64\n",
      " 105  ExAC_AF_FIN                    2137 non-null   float64\n",
      " 106  ExAC_AF_NFE                    2137 non-null   float64\n",
      " 107  ExAC_AF_OTH                    2137 non-null   float64\n",
      " 108  ExAC_AF_SAS                    2137 non-null   float64\n",
      " 109  GENE_PHENO                     0 non-null      float64\n",
      " 110  FILTER                         10449 non-null  object \n",
      " 111  CONTEXT                        10449 non-null  object \n",
      " 112  src_vcf_id                     10449 non-null  object \n",
      " 113  tumor_bam_uuid                 10449 non-null  object \n",
      " 114  normal_bam_uuid                10449 non-null  object \n",
      " 115  case_id                        10449 non-null  object \n",
      " 116  GDC_FILTER                     866 non-null    object \n",
      " 117  COSMIC                         1862 non-null   object \n",
      " 118  MC3_Overlap                    10449 non-null  bool   \n",
      " 119  GDC_Validation_Status          10449 non-null  object \n",
      "dtypes: bool(1), float64(39), int64(10), object(70)\n",
      "memory usage: 9.5+ MB\n"
     ]
    }
   ],
   "source": [
    "table.info(verbose=True,show_counts=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "16e54cda",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10449, 10439, 10416, 10364, 10349, 10347, 10224, 10007, 9970, 9970, 9970, 9970, 9970, 9908, 9848, 9848, 9583, 8701, 8672, 8587, 8312, 8312, 8312, 8312, 8312, 8312, 8312, 8312, 8312, 8040, 7821, 7821, 7821, 4855, 4461, 3906, 2533, 2186, 1447, 1446, 1446, 1315, 1314, 1077, 942, 689, 626, 621, 621, 515, 202, 202, 202, 85, 80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]\n"
     ]
    }
   ],
   "source": [
    "m = table.isna().sum().tolist()\n",
    "m.sort(reverse =True)\n",
    "print(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b3c0d7e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "Hugo_Symbol = table.loc[:,[\"Hugo_Symbol\"]]\n",
    "Y = table.loc[:,[\"Entrez_Gene_Id\"]].values #independent variables\n",
    "X = table.drop([\"Hugo_Symbol\",\"Entrez_Gene_Id\"], axis = 1) #dependent variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7742ed61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[  6536],\n",
       "       [  1263],\n",
       "       [199920],\n",
       "       ...,\n",
       "       [441531],\n",
       "       [  9737],\n",
       "       [114824]], dtype=int64)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Y.tolist()\n",
    "Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6b3bf3ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([['BI', 'GRCh38', 'chr1', ..., 'COSM6063461;COSM6063462', True,\n",
       "        'Valid'],\n",
       "       ['BI', 'GRCh38', 'chr1', ..., nan, True, 'Valid'],\n",
       "       ['BI', 'GRCh38', 'chr1', ..., nan, True, 'Unknown'],\n",
       "       ...,\n",
       "       ['BI', 'GRCh38', 'chrX', ..., nan, True, 'Unknown'],\n",
       "       ['BI', 'GRCh38', 'chrX', ..., nan, True, 'Unknown'],\n",
       "       ['BI', 'GRCh38', 'chrX', ..., nan, True, 'Unknown']], dtype=object)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e5cbe653",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[10439, 10416, 10364, 10349, 10347, 10224, 10007, 9970, 9970, 9970, 9970, 9970, 9908, 9848, 9848, 9583, 8701, 8672, 8587, 8312, 8312, 8312, 8312, 8312, 8312, 8312, 8312, 8312, 8040, 7821, 7821, 7821, 4855, 4461, 3906, 2533, 2186, 1447, 1446, 1446, 1315, 1314, 1077, 942, 689, 626, 621, 621, 515, 202, 202, 202, 85, 80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]\n"
     ]
    }
   ],
   "source": [
    "#Remove columns with all row values NaN\n",
    "X.dropna(axis=1,how=\"all\", inplace = True)\n",
    "m = X.isna().sum().tolist()\n",
    "m.sort(reverse =True)\n",
    "print(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4395db64",
   "metadata": {},
   "outputs": [],
   "source": [
    "#missing_ratio = []\n",
    "#for i in m:\n",
    "#    value = (i/10449)*100\n",
    "#    missing_ratio.append(value)\n",
    "#missing_ratio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8c5c9d3e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Handling NaN in float type cloumns that are below treshold 0.70(70%)#\n",
    "def delete_nan_values(X):\n",
    "    length = len(Y)\n",
    "    thresh = length*0.70\n",
    "    m = X[X.columns[X.dtypes == float]]\n",
    "    m = m.dropna(axis=1,thresh=int(thresh)).fillna(m.mean())\n",
    "    X = X.drop(X.columns[X.dtypes == float], axis=1)\n",
    "    X = pd.concat([X,m], axis=1)\n",
    "    return X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "27a37b96",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Handling categorical values\n",
    "def handle_categ_values(X):\n",
    "    labelencoder_x = LabelEncoder()\n",
    "    m = X.columns[X.dtypes == object]\n",
    "    for n in m:\n",
    "        X[n] = labelencoder_x.fit_transform(X[n])\n",
    "    m = X.columns[X.dtypes == bool]\n",
    "    for n in m:\n",
    "        X[n] = labelencoder_x.fit_transform(X[n])    \n",
    "    return X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d3d4b9f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>...</th>\n",
       "      <th>src_vcf_id</th>\n",
       "      <th>tumor_bam_uuid</th>\n",
       "      <th>normal_bam_uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>GDC_FILTER</th>\n",
       "      <th>COSMIC</th>\n",
       "      <th>MC3_Overlap</th>\n",
       "      <th>GDC_Validation_Status</th>\n",
       "      <th>PICK</th>\n",
       "      <th>TSL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>44008472</td>\n",
       "      <td>44008472</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>1620</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>44803280</td>\n",
       "      <td>44803280</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>64</td>\n",
       "      <td>64</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>56740711</td>\n",
       "      <td>56740711</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>155233116</td>\n",
       "      <td>155233116</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>1825</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>157581530</td>\n",
       "      <td>157581530</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10444</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>18642139</td>\n",
       "      <td>18642139</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>45</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10445</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>47201524</td>\n",
       "      <td>47201524</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>83</td>\n",
       "      <td>83</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>45</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10446</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>77969434</td>\n",
       "      <td>77969434</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>83</td>\n",
       "      <td>83</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>45</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.543215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10447</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>102657224</td>\n",
       "      <td>102657224</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>45</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10448</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>152991432</td>\n",
       "      <td>152991432</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>45</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>1825</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10449 rows Ã— 81 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Center  NCBI_Build  Chromosome  Start_Position  End_Position  Strand  \\\n",
       "0           0           0           0        44008472      44008472       0   \n",
       "1           0           0           0        44803280      44803280       0   \n",
       "2           0           0           0        56740711      56740711       0   \n",
       "3           0           0           0       155233116     155233116       0   \n",
       "4           0           0           0       157581530     157581530       0   \n",
       "...       ...         ...         ...             ...           ...     ...   \n",
       "10444       0           0          22        18642139      18642139       0   \n",
       "10445       0           0          22        47201524      47201524       0   \n",
       "10446       0           0          22        77969434      77969434       0   \n",
       "10447       0           0          22       102657224     102657224       0   \n",
       "10448       0           0          22       152991432     152991432       0   \n",
       "\n",
       "       Variant_Classification  Variant_Type  Reference_Allele  \\\n",
       "0                          13             2                64   \n",
       "1                           9             2                64   \n",
       "2                          13             2                31   \n",
       "3                           0             2                31   \n",
       "4                          15             2                31   \n",
       "...                       ...           ...               ...   \n",
       "10444                      13             2                31   \n",
       "10445                       9             2                83   \n",
       "10446                       9             2                83   \n",
       "10447                       9             2                31   \n",
       "10448                       9             2                 1   \n",
       "\n",
       "       Tumor_Seq_Allele1  ...  src_vcf_id  tumor_bam_uuid  normal_bam_uuid  \\\n",
       "0                     64  ...          31               7               20   \n",
       "1                     64  ...          31               7               20   \n",
       "2                     31  ...          31               7               20   \n",
       "3                     31  ...          31               7               20   \n",
       "4                     31  ...          31               7               20   \n",
       "...                  ...  ...         ...             ...              ...   \n",
       "10444                 31  ...           5              26               45   \n",
       "10445                 83  ...           5              26               45   \n",
       "10446                 83  ...           5              26               45   \n",
       "10447                 31  ...           5              26               45   \n",
       "10448                  1  ...           5              26               45   \n",
       "\n",
       "       case_id  GDC_FILTER  COSMIC  MC3_Overlap  GDC_Validation_Status  PICK  \\\n",
       "0            1           8    1620            1                      1   1.0   \n",
       "1            1           8    1825            1                      1   1.0   \n",
       "2            1           5    1825            1                      0   1.0   \n",
       "3            1           8    1825            0                      0   1.0   \n",
       "4            1           0    1825            1                      0   1.0   \n",
       "...        ...         ...     ...          ...                    ...   ...   \n",
       "10444        5           8    1825            1                      0   1.0   \n",
       "10445        5           8    1825            1                      0   1.0   \n",
       "10446        5           8    1825            1                      0   1.0   \n",
       "10447        5           8    1825            1                      0   1.0   \n",
       "10448        5           8    1825            1                      0   1.0   \n",
       "\n",
       "            TSL  \n",
       "0      1.000000  \n",
       "1      1.000000  \n",
       "2      1.000000  \n",
       "3      1.000000  \n",
       "4      1.000000  \n",
       "...         ...  \n",
       "10444  1.000000  \n",
       "10445  1.000000  \n",
       "10446  1.543215  \n",
       "10447  2.000000  \n",
       "10448  1.000000  \n",
       "\n",
       "[10449 rows x 81 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X=delete_nan_values(X)\n",
    "X=handle_categ_values(X)\n",
    "X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "6c9b25f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Data columns (total 81 columns):\n",
      " #   Column                       Non-Null Count  Dtype  \n",
      "---  ------                       --------------  -----  \n",
      " 0   Center                       10449 non-null  int32  \n",
      " 1   NCBI_Build                   10449 non-null  int32  \n",
      " 2   Chromosome                   10449 non-null  int32  \n",
      " 3   Start_Position               10449 non-null  int64  \n",
      " 4   End_Position                 10449 non-null  int64  \n",
      " 5   Strand                       10449 non-null  int32  \n",
      " 6   Variant_Classification       10449 non-null  int32  \n",
      " 7   Variant_Type                 10449 non-null  int32  \n",
      " 8   Reference_Allele             10449 non-null  int32  \n",
      " 9   Tumor_Seq_Allele1            10449 non-null  int32  \n",
      " 10  Tumor_Seq_Allele2            10449 non-null  int32  \n",
      " 11  dbSNP_RS                     10449 non-null  int32  \n",
      " 12  dbSNP_Val_Status             10449 non-null  int32  \n",
      " 13  Tumor_Sample_Barcode         10449 non-null  int32  \n",
      " 14  Matched_Norm_Sample_Barcode  10449 non-null  int32  \n",
      " 15  Tumor_Validation_Allele1     10449 non-null  int32  \n",
      " 16  Tumor_Validation_Allele2     10449 non-null  int32  \n",
      " 17  Mutation_Status              10449 non-null  int32  \n",
      " 18  Validation_Method            10449 non-null  int32  \n",
      " 19  Sequencer                    10449 non-null  int32  \n",
      " 20  Tumor_Sample_UUID            10449 non-null  int32  \n",
      " 21  Matched_Norm_Sample_UUID     10449 non-null  int32  \n",
      " 22  HGVSc                        10449 non-null  int32  \n",
      " 23  HGVSp                        10449 non-null  int32  \n",
      " 24  HGVSp_Short                  10449 non-null  int32  \n",
      " 25  Transcript_ID                10449 non-null  int32  \n",
      " 26  Exon_Number                  10449 non-null  int32  \n",
      " 27  t_depth                      10449 non-null  int64  \n",
      " 28  t_ref_count                  10449 non-null  int64  \n",
      " 29  t_alt_count                  10449 non-null  int64  \n",
      " 30  n_depth                      10449 non-null  int64  \n",
      " 31  all_effects                  10449 non-null  int32  \n",
      " 32  Allele                       10449 non-null  int32  \n",
      " 33  Gene                         10449 non-null  int32  \n",
      " 34  Feature                      10449 non-null  int32  \n",
      " 35  Feature_type                 10449 non-null  int32  \n",
      " 36  One_Consequence              10449 non-null  int32  \n",
      " 37  Consequence                  10449 non-null  int32  \n",
      " 38  cDNA_position                10449 non-null  int32  \n",
      " 39  CDS_position                 10449 non-null  int32  \n",
      " 40  Protein_position             10449 non-null  int32  \n",
      " 41  Amino_acids                  10449 non-null  int32  \n",
      " 42  Codons                       10449 non-null  int32  \n",
      " 43  Existing_variation           10449 non-null  int32  \n",
      " 44  ALLELE_NUM                   10449 non-null  int64  \n",
      " 45  TRANSCRIPT_STRAND            10449 non-null  int64  \n",
      " 46  SYMBOL                       10449 non-null  int32  \n",
      " 47  SYMBOL_SOURCE                10449 non-null  int32  \n",
      " 48  HGNC_ID                      10449 non-null  int32  \n",
      " 49  BIOTYPE                      10449 non-null  int32  \n",
      " 50  CANONICAL                    10449 non-null  int32  \n",
      " 51  CCDS                         10449 non-null  int32  \n",
      " 52  ENSP                         10449 non-null  int32  \n",
      " 53  SWISSPROT                    10449 non-null  int32  \n",
      " 54  TREMBL                       10449 non-null  int32  \n",
      " 55  UNIPARC                      10449 non-null  int32  \n",
      " 56  RefSeq                       10449 non-null  int32  \n",
      " 57  SIFT                         10449 non-null  int32  \n",
      " 58  PolyPhen                     10449 non-null  int32  \n",
      " 59  EXON                         10449 non-null  int32  \n",
      " 60  INTRON                       10449 non-null  int32  \n",
      " 61  DOMAINS                      10449 non-null  int32  \n",
      " 62  CLIN_SIG                     10449 non-null  int32  \n",
      " 63  SOMATIC                      10449 non-null  int32  \n",
      " 64  PUBMED                       10449 non-null  int32  \n",
      " 65  IMPACT                       10449 non-null  int32  \n",
      " 66  VARIANT_CLASS                10449 non-null  int32  \n",
      " 67  PHENO                        10449 non-null  int32  \n",
      " 68  MINIMISED                    10449 non-null  int64  \n",
      " 69  FILTER                       10449 non-null  int32  \n",
      " 70  CONTEXT                      10449 non-null  int32  \n",
      " 71  src_vcf_id                   10449 non-null  int32  \n",
      " 72  tumor_bam_uuid               10449 non-null  int32  \n",
      " 73  normal_bam_uuid              10449 non-null  int32  \n",
      " 74  case_id                      10449 non-null  int32  \n",
      " 75  GDC_FILTER                   10449 non-null  int32  \n",
      " 76  COSMIC                       10449 non-null  int32  \n",
      " 77  MC3_Overlap                  10449 non-null  int64  \n",
      " 78  GDC_Validation_Status        10449 non-null  int32  \n",
      " 79  PICK                         10449 non-null  float64\n",
      " 80  TSL                          10449 non-null  float64\n",
      "dtypes: float64(2), int32(69), int64(10)\n",
      "memory usage: 3.7 MB\n"
     ]
    }
   ],
   "source": [
    "X.info(verbose=True,show_counts=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "4805b1f9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]\n"
     ]
    }
   ],
   "source": [
    "m = X.isna().sum().tolist()\n",
    "m.sort(reverse =True)\n",
    "print(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "586b61a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\anjel\\anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py:63: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().\n",
      "  return f(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([1695,  344, 6058, ..., 6697, 2491, 5458], dtype=int64)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labelencoder_y= LabelEncoder()  \n",
    "Y = labelencoder_y.fit_transform(Y)\n",
    "Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "efccfcd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(X.values, Y, test_size = 0.30,random_state=42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "28d8a0f4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 0.,  0.,  7., ...,  0.,  1.,  1.],\n",
       "       [ 0.,  0.,  7., ...,  0.,  1.,  5.],\n",
       "       [ 0.,  0., 19., ...,  1.,  1.,  1.],\n",
       "       ...,\n",
       "       [ 0.,  0., 15., ...,  0.,  1.,  1.],\n",
       "       [ 0.,  0., 11., ...,  0.,  1.,  1.],\n",
       "       [ 0.,  0.,  2., ...,  0.,  1.,  1.]])"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X_train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "3b3b5711",
   "metadata": {},
   "outputs": [],
   "source": [
    "st_x= StandardScaler()\n",
    "X_train= st_x.fit_transform(X_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d04e7af8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 0.        ,  0.        , -0.51498828, ..., -0.58229712,\n",
       "         0.        , -0.43041155],\n",
       "       [ 0.        ,  0.        , -0.51498828, ..., -0.58229712,\n",
       "         0.        ,  2.80988519],\n",
       "       [ 0.        ,  0.        ,  1.15375171, ...,  1.71733633,\n",
       "         0.        , -0.43041155],\n",
       "       ...,\n",
       "       [ 0.        ,  0.        ,  0.59750504, ..., -0.58229712,\n",
       "         0.        , -0.43041155],\n",
       "       [ 0.        ,  0.        ,  0.04125838, ..., -0.58229712,\n",
       "         0.        , -0.43041155],\n",
       "       [ 0.        ,  0.        , -1.21029661, ..., -0.58229712,\n",
       "         0.        , -0.43041155]])"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X_train"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "add86046",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
