{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6f74605b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<gzip _io.BufferedReader name='SNV/02747363-f04a-4ba6-a079-fe4f87853788/TCGA.UCS.mutect.02747363-f04a-4ba6-a079-fe4f87853788.DR-10.0.somatic.maf.gz' 0x26864a757c0>\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import time\n",
    "import pandas as pd\n",
    "from sklearn import preprocessing\n",
    "from sklearn.impute import SimpleImputer\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.preprocessing import Normalizer\n",
    "from sklearn.preprocessing import OneHotEncoder\n",
    "\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import classification_report,confusion_matrix\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import gzip\n",
    "\n",
    "df = pd.read_table(\"SNV/MANIFEST.txt\")\n",
    "fileslist = []\n",
    "header = []\n",
    "for file_name in df[0:33][\"filename\"]:\n",
    "    fileslist.append(file_name)\n",
    "with gzip.open(\"SNV/\"+fileslist[0],\"rb\") as f:\n",
    "    print(f)\n",
    "    i = 0\n",
    "    while i <=4:\n",
    "        header.append(f.readline())\n",
    "        i += 1\n",
    "    mf = pd.read_csv(f, delimiter = \"\\t\")\n",
    "    table = mf[0:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "da8d0dcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>src_vcf_id</th>\n",
       "      <th>tumor_bam_uuid</th>\n",
       "      <th>normal_bam_uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>GDC_FILTER</th>\n",
       "      <th>COSMIC</th>\n",
       "      <th>MC3_Overlap</th>\n",
       "      <th>GDC_Validation_Status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SLC6A9</td>\n",
       "      <td>6536</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44008472</td>\n",
       "      <td>44008472</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ACACCGGCGCA</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COSM6063461;COSM6063462</td>\n",
       "      <td>True</td>\n",
       "      <td>Valid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PLK3</td>\n",
       "      <td>1263</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44803280</td>\n",
       "      <td>44803280</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CCCCCGATCGA</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Valid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C1orf168</td>\n",
       "      <td>199920</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56740711</td>\n",
       "      <td>56740711</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TCTTTCGACTT</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>common_in_exac</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GBA</td>\n",
       "      <td>2629</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>155233116</td>\n",
       "      <td>155233116</td>\n",
       "      <td>+</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GCATGCAGCGG</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FCRL4</td>\n",
       "      <td>83417</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>157581530</td>\n",
       "      <td>157581530</td>\n",
       "      <td>+</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ACAAACCTGAC</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NonExonic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10444</th>\n",
       "      <td>RS1</td>\n",
       "      <td>6247</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>18642139</td>\n",
       "      <td>18642139</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CGGTCCGAGTT</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10445</th>\n",
       "      <td>UBA1</td>\n",
       "      <td>7317</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>47201524</td>\n",
       "      <td>47201524</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CGGGTTTGAGA</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10446</th>\n",
       "      <td>PGAM4</td>\n",
       "      <td>441531</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>77969434</td>\n",
       "      <td>77969434</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CACTGTCCAGA</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10447</th>\n",
       "      <td>GPRASP1</td>\n",
       "      <td>9737</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>102657224</td>\n",
       "      <td>102657224</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TCAGCCTGATC</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10448</th>\n",
       "      <td>PNMA5</td>\n",
       "      <td>114824</td>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>152991432</td>\n",
       "      <td>152991432</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TCCTGAACATT</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10449 rows × 120 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  \\\n",
       "0          SLC6A9            6536     BI     GRCh38       chr1   \n",
       "1            PLK3            1263     BI     GRCh38       chr1   \n",
       "2        C1orf168          199920     BI     GRCh38       chr1   \n",
       "3             GBA            2629     BI     GRCh38       chr1   \n",
       "4           FCRL4           83417     BI     GRCh38       chr1   \n",
       "...           ...             ...    ...        ...        ...   \n",
       "10444         RS1            6247     BI     GRCh38       chrX   \n",
       "10445        UBA1            7317     BI     GRCh38       chrX   \n",
       "10446       PGAM4          441531     BI     GRCh38       chrX   \n",
       "10447     GPRASP1            9737     BI     GRCh38       chrX   \n",
       "10448       PNMA5          114824     BI     GRCh38       chrX   \n",
       "\n",
       "       Start_Position  End_Position Strand Variant_Classification  \\\n",
       "0            44008472      44008472      +                 Silent   \n",
       "1            44803280      44803280      +      Missense_Mutation   \n",
       "2            56740711      56740711      +                 Silent   \n",
       "3           155233116     155233116      +                3'Flank   \n",
       "4           157581530     157581530      +            Splice_Site   \n",
       "...               ...           ...    ...                    ...   \n",
       "10444        18642139      18642139      +                 Silent   \n",
       "10445        47201524      47201524      +      Missense_Mutation   \n",
       "10446        77969434      77969434      +      Missense_Mutation   \n",
       "10447       102657224     102657224      +      Missense_Mutation   \n",
       "10448       152991432     152991432      +      Missense_Mutation   \n",
       "\n",
       "      Variant_Type  ... FILTER      CONTEXT  \\\n",
       "0              SNP  ...   PASS  ACACCGGCGCA   \n",
       "1              SNP  ...   PASS  CCCCCGATCGA   \n",
       "2              SNP  ...   PASS  TCTTTCGACTT   \n",
       "3              SNP  ...   PASS  GCATGCAGCGG   \n",
       "4              SNP  ...   PASS  ACAAACCTGAC   \n",
       "...            ...  ...    ...          ...   \n",
       "10444          SNP  ...   PASS  CGGTCCGAGTT   \n",
       "10445          SNP  ...   PASS  CGGGTTTGAGA   \n",
       "10446          SNP  ...   PASS  CACTGTCCAGA   \n",
       "10447          SNP  ...   PASS  TCAGCCTGATC   \n",
       "10448          SNP  ...   PASS  TCCTGAACATT   \n",
       "\n",
       "                                 src_vcf_id  \\\n",
       "0      85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "1      85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "2      85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "3      85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "4      85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "...                                     ...   \n",
       "10444  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10445  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10446  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10447  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10448  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "\n",
       "                             tumor_bam_uuid  \\\n",
       "0      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "1      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "2      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "3      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "4      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "...                                     ...   \n",
       "10444  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10445  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10446  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10447  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10448  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "\n",
       "                            normal_bam_uuid  \\\n",
       "0      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "1      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "2      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "3      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "4      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "...                                     ...   \n",
       "10444  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10445  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10446  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10447  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10448  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "\n",
       "                                    case_id      GDC_FILTER  \\\n",
       "0      14213209-2217-4812-9a19-d9b2b6718467             NaN   \n",
       "1      14213209-2217-4812-9a19-d9b2b6718467             NaN   \n",
       "2      14213209-2217-4812-9a19-d9b2b6718467  common_in_exac   \n",
       "3      14213209-2217-4812-9a19-d9b2b6718467             NaN   \n",
       "4      14213209-2217-4812-9a19-d9b2b6718467       NonExonic   \n",
       "...                                     ...             ...   \n",
       "10444  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10445  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10446  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10447  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10448  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "\n",
       "                        COSMIC  MC3_Overlap GDC_Validation_Status  \n",
       "0      COSM6063461;COSM6063462         True                 Valid  \n",
       "1                          NaN         True                 Valid  \n",
       "2                          NaN         True               Unknown  \n",
       "3                          NaN        False               Unknown  \n",
       "4                          NaN         True               Unknown  \n",
       "...                        ...          ...                   ...  \n",
       "10444                      NaN         True               Unknown  \n",
       "10445                      NaN         True               Unknown  \n",
       "10446                      NaN         True               Unknown  \n",
       "10447                      NaN         True               Unknown  \n",
       "10448                      NaN         True               Unknown  \n",
       "\n",
       "[10449 rows x 120 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f4a8436d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Entrez_Gene_Id\n",
      "0                6536\n",
      "1                1263\n",
      "2              199920\n",
      "3                2629\n",
      "4               83417\n",
      "...               ...\n",
      "10444            6247\n",
      "10445            7317\n",
      "10446          441531\n",
      "10447            9737\n",
      "10448          114824\n",
      "\n",
      "[10449 rows x 1 columns]\n",
      "      Hugo_Symbol\n",
      "0          SLC6A9\n",
      "1            PLK3\n",
      "2        C1orf168\n",
      "3             GBA\n",
      "4           FCRL4\n",
      "...           ...\n",
      "10444         RS1\n",
      "10445        UBA1\n",
      "10446       PGAM4\n",
      "10447     GPRASP1\n",
      "10448       PNMA5\n",
      "\n",
      "[10449 rows x 1 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>...</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>src_vcf_id</th>\n",
       "      <th>tumor_bam_uuid</th>\n",
       "      <th>normal_bam_uuid</th>\n",
       "      <th>case_id</th>\n",
       "      <th>GDC_FILTER</th>\n",
       "      <th>COSMIC</th>\n",
       "      <th>MC3_Overlap</th>\n",
       "      <th>GDC_Validation_Status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44008472</td>\n",
       "      <td>44008472</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ACACCGGCGCA</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>COSM6063461;COSM6063462</td>\n",
       "      <td>True</td>\n",
       "      <td>Valid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44803280</td>\n",
       "      <td>44803280</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CCCCCGATCGA</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Valid</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>56740711</td>\n",
       "      <td>56740711</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TCTTTCGACTT</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>common_in_exac</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>155233116</td>\n",
       "      <td>155233116</td>\n",
       "      <td>+</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>GCATGCAGCGG</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>157581530</td>\n",
       "      <td>157581530</td>\n",
       "      <td>+</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>ACAAACCTGAC</td>\n",
       "      <td>85b191fa-e8f6-4d97-bc13-0bc236cee6c8</td>\n",
       "      <td>3d52eda9-5f36-4a55-8a3c-07a576a0ef1f</td>\n",
       "      <td>6c3037f4-9838-4b36-9551-99e91cb85ef0</td>\n",
       "      <td>14213209-2217-4812-9a19-d9b2b6718467</td>\n",
       "      <td>NonExonic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10444</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>18642139</td>\n",
       "      <td>18642139</td>\n",
       "      <td>+</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CGGTCCGAGTT</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10445</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>47201524</td>\n",
       "      <td>47201524</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CGGGTTTGAGA</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10446</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>77969434</td>\n",
       "      <td>77969434</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>CACTGTCCAGA</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10447</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>102657224</td>\n",
       "      <td>102657224</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TCAGCCTGATC</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10448</th>\n",
       "      <td>BI</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>chrX</td>\n",
       "      <td>152991432</td>\n",
       "      <td>152991432</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TCCTGAACATT</td>\n",
       "      <td>1d4f6015-1138-4d11-ae68-5034b9af7511</td>\n",
       "      <td>810c44b5-8901-4877-8ca6-bedfd040e0c2</td>\n",
       "      <td>e052d1b5-8959-43df-916a-8dce7977fdc0</td>\n",
       "      <td>200f77a5-803c-4891-980b-5c3b08c91254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10449 rows × 118 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Center NCBI_Build Chromosome  Start_Position  End_Position Strand  \\\n",
       "0         BI     GRCh38       chr1        44008472      44008472      +   \n",
       "1         BI     GRCh38       chr1        44803280      44803280      +   \n",
       "2         BI     GRCh38       chr1        56740711      56740711      +   \n",
       "3         BI     GRCh38       chr1       155233116     155233116      +   \n",
       "4         BI     GRCh38       chr1       157581530     157581530      +   \n",
       "...      ...        ...        ...             ...           ...    ...   \n",
       "10444     BI     GRCh38       chrX        18642139      18642139      +   \n",
       "10445     BI     GRCh38       chrX        47201524      47201524      +   \n",
       "10446     BI     GRCh38       chrX        77969434      77969434      +   \n",
       "10447     BI     GRCh38       chrX       102657224     102657224      +   \n",
       "10448     BI     GRCh38       chrX       152991432     152991432      +   \n",
       "\n",
       "      Variant_Classification Variant_Type Reference_Allele Tumor_Seq_Allele1  \\\n",
       "0                     Silent          SNP                G                 G   \n",
       "1          Missense_Mutation          SNP                G                 G   \n",
       "2                     Silent          SNP                C                 C   \n",
       "3                    3'Flank          SNP                C                 C   \n",
       "4                Splice_Site          SNP                C                 C   \n",
       "...                      ...          ...              ...               ...   \n",
       "10444                 Silent          SNP                C                 C   \n",
       "10445      Missense_Mutation          SNP                T                 T   \n",
       "10446      Missense_Mutation          SNP                T                 T   \n",
       "10447      Missense_Mutation          SNP                C                 C   \n",
       "10448      Missense_Mutation          SNP                A                 A   \n",
       "\n",
       "       ... FILTER      CONTEXT                            src_vcf_id  \\\n",
       "0      ...   PASS  ACACCGGCGCA  85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "1      ...   PASS  CCCCCGATCGA  85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "2      ...   PASS  TCTTTCGACTT  85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "3      ...   PASS  GCATGCAGCGG  85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "4      ...   PASS  ACAAACCTGAC  85b191fa-e8f6-4d97-bc13-0bc236cee6c8   \n",
       "...    ...    ...          ...                                   ...   \n",
       "10444  ...   PASS  CGGTCCGAGTT  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10445  ...   PASS  CGGGTTTGAGA  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10446  ...   PASS  CACTGTCCAGA  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10447  ...   PASS  TCAGCCTGATC  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "10448  ...   PASS  TCCTGAACATT  1d4f6015-1138-4d11-ae68-5034b9af7511   \n",
       "\n",
       "                             tumor_bam_uuid  \\\n",
       "0      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "1      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "2      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "3      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "4      3d52eda9-5f36-4a55-8a3c-07a576a0ef1f   \n",
       "...                                     ...   \n",
       "10444  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10445  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10446  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10447  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "10448  810c44b5-8901-4877-8ca6-bedfd040e0c2   \n",
       "\n",
       "                            normal_bam_uuid  \\\n",
       "0      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "1      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "2      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "3      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "4      6c3037f4-9838-4b36-9551-99e91cb85ef0   \n",
       "...                                     ...   \n",
       "10444  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10445  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10446  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10447  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "10448  e052d1b5-8959-43df-916a-8dce7977fdc0   \n",
       "\n",
       "                                    case_id      GDC_FILTER  \\\n",
       "0      14213209-2217-4812-9a19-d9b2b6718467             NaN   \n",
       "1      14213209-2217-4812-9a19-d9b2b6718467             NaN   \n",
       "2      14213209-2217-4812-9a19-d9b2b6718467  common_in_exac   \n",
       "3      14213209-2217-4812-9a19-d9b2b6718467             NaN   \n",
       "4      14213209-2217-4812-9a19-d9b2b6718467       NonExonic   \n",
       "...                                     ...             ...   \n",
       "10444  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10445  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10446  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10447  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "10448  200f77a5-803c-4891-980b-5c3b08c91254             NaN   \n",
       "\n",
       "                        COSMIC MC3_Overlap  GDC_Validation_Status  \n",
       "0      COSM6063461;COSM6063462        True                  Valid  \n",
       "1                          NaN        True                  Valid  \n",
       "2                          NaN        True                Unknown  \n",
       "3                          NaN       False                Unknown  \n",
       "4                          NaN        True                Unknown  \n",
       "...                        ...         ...                    ...  \n",
       "10444                      NaN        True                Unknown  \n",
       "10445                      NaN        True                Unknown  \n",
       "10446                      NaN        True                Unknown  \n",
       "10447                      NaN        True                Unknown  \n",
       "10448                      NaN        True                Unknown  \n",
       "\n",
       "[10449 rows x 118 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Hugo_Symbol = table.loc[:,[\"Hugo_Symbol\"]]\n",
    "Y = table.loc[:,[\"Entrez_Gene_Id\"]] #independent variables\n",
    "print(Y)\n",
    "print(Hugo_Symbol)\n",
    "X = table.drop([\"Hugo_Symbol\",\"Entrez_Gene_Id\"], axis = 1) #dependent variables\n",
    "X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b1ea3e3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "cb5a8c33",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "54\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "a6f4b3c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Handle features and genes_id#\n",
    "\n",
    "#Retrieve Y (genes_id) and X (all other features)\n",
    "Y = new_table.loc[:,[\"Hugo_Symbol\"]].values\n",
    "length = len(Y)\n",
    "\n",
    "#Retrive features\n",
    "X = new_table.drop([\"Hugo_Symbol\"], axis = 1)\n",
    "thresh = length*0.90"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "6f076c7e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Data columns (total 101 columns):\n",
      " #    Column                       Non-Null Count  Dtype  \n",
      "---   ------                       --------------  -----  \n",
      " 0    Entrez_Gene_Id               10449 non-null  int64  \n",
      " 1    Center                       10449 non-null  object \n",
      " 2    NCBI_Build                   10449 non-null  object \n",
      " 3    Chromosome                   10449 non-null  object \n",
      " 4    Start_Position               10449 non-null  int64  \n",
      " 5    End_Position                 10449 non-null  int64  \n",
      " 6    Strand                       10449 non-null  object \n",
      " 7    Variant_Classification       10449 non-null  object \n",
      " 8    Variant_Type                 10449 non-null  object \n",
      " 9    Reference_Allele             10449 non-null  object \n",
      " 10   Tumor_Seq_Allele1            10449 non-null  object \n",
      " 11   Tumor_Seq_Allele2            10449 non-null  object \n",
      " 12   dbSNP_RS                     9507 non-null   object \n",
      " 13   dbSNP_Val_Status             1777 non-null   object \n",
      " 14   Tumor_Sample_Barcode         10449 non-null  object \n",
      " 15   Matched_Norm_Sample_Barcode  10449 non-null  object \n",
      " 16   Tumor_Validation_Allele1     2628 non-null   object \n",
      " 17   Tumor_Validation_Allele2     2628 non-null   object \n",
      " 18   Mutation_Status              10449 non-null  object \n",
      " 19   Validation_Method            2628 non-null   object \n",
      " 20   Sequencer                    10449 non-null  object \n",
      " 21   Tumor_Sample_UUID            10449 non-null  object \n",
      " 22   Matched_Norm_Sample_UUID     10449 non-null  object \n",
      " 23   HGVSc                        10364 non-null  object \n",
      " 24   HGVSp                        9002 non-null   object \n",
      " 25   HGVSp_Short                  9134 non-null   object \n",
      " 26   Transcript_ID                10449 non-null  object \n",
      " 27   Exon_Number                  9828 non-null   object \n",
      " 28   t_depth                      10449 non-null  int64  \n",
      " 29   t_ref_count                  10449 non-null  int64  \n",
      " 30   t_alt_count                  10449 non-null  int64  \n",
      " 31   n_depth                      10449 non-null  int64  \n",
      " 32   all_effects                  10449 non-null  object \n",
      " 33   Allele                       10449 non-null  object \n",
      " 34   Gene                         10449 non-null  object \n",
      " 35   Feature                      10449 non-null  object \n",
      " 36   Feature_type                 10449 non-null  object \n",
      " 37   One_Consequence              10449 non-null  object \n",
      " 38   Consequence                  10449 non-null  object \n",
      " 39   cDNA_position                10449 non-null  object \n",
      " 40   CDS_position                 10247 non-null  object \n",
      " 41   Protein_position             10247 non-null  object \n",
      " 42   Amino_acids                  9003 non-null   object \n",
      " 43   Codons                       9003 non-null   object \n",
      " 44   Existing_variation           2409 non-null   object \n",
      " 45   ALLELE_NUM                   10449 non-null  int64  \n",
      " 46   DISTANCE                     85 non-null     float64\n",
      " 47   TRANSCRIPT_STRAND            10449 non-null  int64  \n",
      " 48   SYMBOL                       10449 non-null  object \n",
      " 49   SYMBOL_SOURCE                10449 non-null  object \n",
      " 50   HGNC_ID                      10369 non-null  object \n",
      " 51   BIOTYPE                      10449 non-null  object \n",
      " 52   CANONICAL                    8263 non-null   object \n",
      " 53   CCDS                         9760 non-null   object \n",
      " 54   ENSP                         10247 non-null  object \n",
      " 55   SWISSPROT                    9934 non-null   object \n",
      " 56   TREMBL                       1748 non-null   object \n",
      " 57   UNIPARC                      9135 non-null   object \n",
      " 58   RefSeq                       9372 non-null   object \n",
      " 59   SIFT                         5594 non-null   object \n",
      " 60   PolyPhen                     5988 non-null   object \n",
      " 61   EXON                         9828 non-null   object \n",
      " 62   INTRON                       541 non-null    object \n",
      " 63   DOMAINS                      6543 non-null   object \n",
      " 64   GMAF                         442 non-null    float64\n",
      " 65   AFR_MAF                      479 non-null    float64\n",
      " 66   AMR_MAF                      479 non-null    float64\n",
      " 67   EAS_MAF                      479 non-null    float64\n",
      " 68   EUR_MAF                      479 non-null    float64\n",
      " 69   SAS_MAF                      479 non-null    float64\n",
      " 70   AA_MAF                       601 non-null    float64\n",
      " 71   EA_MAF                       601 non-null    float64\n",
      " 72   CLIN_SIG                     100 non-null    object \n",
      " 73   SOMATIC                      10 non-null     object \n",
      " 74   PUBMED                       33 non-null     object \n",
      " 75   IMPACT                       10449 non-null  object \n",
      " 76   PICK                         7916 non-null   float64\n",
      " 77   VARIANT_CLASS                10449 non-null  object \n",
      " 78   TSL                          9823 non-null   float64\n",
      " 79   HGVS_OFFSET                  225 non-null    float64\n",
      " 80   PHENO                        102 non-null    object \n",
      " 81   MINIMISED                    10449 non-null  int64  \n",
      " 82   ExAC_AF                      2137 non-null   float64\n",
      " 83   ExAC_AF_Adj                  2137 non-null   float64\n",
      " 84   ExAC_AF_AFR                  2137 non-null   float64\n",
      " 85   ExAC_AF_AMR                  2137 non-null   float64\n",
      " 86   ExAC_AF_EAS                  2137 non-null   float64\n",
      " 87   ExAC_AF_FIN                  2137 non-null   float64\n",
      " 88   ExAC_AF_NFE                  2137 non-null   float64\n",
      " 89   ExAC_AF_OTH                  2137 non-null   float64\n",
      " 90   ExAC_AF_SAS                  2137 non-null   float64\n",
      " 91   FILTER                       10449 non-null  object \n",
      " 92   CONTEXT                      10449 non-null  object \n",
      " 93   src_vcf_id                   10449 non-null  object \n",
      " 94   tumor_bam_uuid               10449 non-null  object \n",
      " 95   normal_bam_uuid              10449 non-null  object \n",
      " 96   case_id                      10449 non-null  object \n",
      " 97   GDC_FILTER                   866 non-null    object \n",
      " 98   COSMIC                       1862 non-null   object \n",
      " 99   MC3_Overlap                  10449 non-null  bool   \n",
      " 100  GDC_Validation_Status        10449 non-null  object \n",
      "dtypes: bool(1), float64(21), int64(10), object(69)\n",
      "memory usage: 8.0+ MB\n"
     ]
    }
   ],
   "source": [
    "#Handling NaN in cloumns below treshold 0.90(90%)#\n",
    "#X  = X.dropna(axis=1,thresh=int(thresh))\n",
    "#X.isna().sum().sort_values(ascending=False)\n",
    "X.info(verbose=True,show_counts=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "c1d9093c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Columns: 101 entries, Entrez_Gene_Id to GDC_Validation_Status\n",
      "dtypes: bool(1), float64(21), int64(10), object(69)\n",
      "memory usage: 8.0+ MB\n",
      "17\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>GMAF</th>\n",
       "      <th>AFR_MAF</th>\n",
       "      <th>AMR_MAF</th>\n",
       "      <th>EAS_MAF</th>\n",
       "      <th>EUR_MAF</th>\n",
       "      <th>SAS_MAF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>GDC_FILTER</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.262)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.262)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>common_in_exac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1336.0</td>\n",
       "      <td>A0A068F658</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.262)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>common_in_exac</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1336.0</td>\n",
       "      <td>A0A068F658</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.262)</td>\n",
       "      <td>7/11</td>\n",
       "      <td>0.0002</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NonExonic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10444</th>\n",
       "      <td>2205.0</td>\n",
       "      <td>A0A024R0J1</td>\n",
       "      <td>tolerated(0.32)</td>\n",
       "      <td>benign(0.026)</td>\n",
       "      <td>9/22</td>\n",
       "      <td>0.0042</td>\n",
       "      <td>0.0159</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>likely_pathogenic;pathogenic</td>\n",
       "      <td>1;0</td>\n",
       "      <td>25105660</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1;0</td>\n",
       "      <td>common_in_exac;gdc_pon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10445</th>\n",
       "      <td>2205.0</td>\n",
       "      <td>A0A024R1A3</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.917)</td>\n",
       "      <td>9/22</td>\n",
       "      <td>0.0042</td>\n",
       "      <td>0.0159</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>likely_pathogenic;pathogenic</td>\n",
       "      <td>1;0</td>\n",
       "      <td>25105660</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1;0</td>\n",
       "      <td>common_in_exac;gdc_pon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10446</th>\n",
       "      <td>2205.0</td>\n",
       "      <td>A0A024R1A3</td>\n",
       "      <td>tolerated(0.41)</td>\n",
       "      <td>benign(0.075)</td>\n",
       "      <td>9/22</td>\n",
       "      <td>0.0042</td>\n",
       "      <td>0.0159</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>likely_pathogenic;pathogenic</td>\n",
       "      <td>1;0</td>\n",
       "      <td>25105660</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1;0</td>\n",
       "      <td>common_in_exac;gdc_pon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10447</th>\n",
       "      <td>2205.0</td>\n",
       "      <td>A0A024R1A3</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>possibly_damaging(0.754)</td>\n",
       "      <td>9/22</td>\n",
       "      <td>0.0042</td>\n",
       "      <td>0.0159</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>likely_pathogenic;pathogenic</td>\n",
       "      <td>1;0</td>\n",
       "      <td>25105660</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1;0</td>\n",
       "      <td>common_in_exac;gdc_pon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10448</th>\n",
       "      <td>2205.0</td>\n",
       "      <td>A0A024R1A3</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.972)</td>\n",
       "      <td>9/22</td>\n",
       "      <td>0.0042</td>\n",
       "      <td>0.0159</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000</td>\n",
       "      <td>likely_pathogenic;pathogenic</td>\n",
       "      <td>1;0</td>\n",
       "      <td>25105660</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1;0</td>\n",
       "      <td>common_in_exac;gdc_pon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10449 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       DISTANCE      TREMBL               SIFT                  PolyPhen  \\\n",
       "0           NaN         NaN                NaN                       NaN   \n",
       "1           NaN         NaN  deleterious(0.01)             benign(0.262)   \n",
       "2           NaN         NaN  deleterious(0.01)             benign(0.262)   \n",
       "3        1336.0  A0A068F658  deleterious(0.01)             benign(0.262)   \n",
       "4        1336.0  A0A068F658  deleterious(0.01)             benign(0.262)   \n",
       "...         ...         ...                ...                       ...   \n",
       "10444    2205.0  A0A024R0J1    tolerated(0.32)             benign(0.026)   \n",
       "10445    2205.0  A0A024R1A3     deleterious(0)  probably_damaging(0.917)   \n",
       "10446    2205.0  A0A024R1A3    tolerated(0.41)             benign(0.075)   \n",
       "10447    2205.0  A0A024R1A3     deleterious(0)  possibly_damaging(0.754)   \n",
       "10448    2205.0  A0A024R1A3     deleterious(0)  probably_damaging(0.972)   \n",
       "\n",
       "      INTRON    GMAF  AFR_MAF  AMR_MAF  EAS_MAF  EUR_MAF  SAS_MAF  \\\n",
       "0        NaN     NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "1        NaN     NaN      NaN      NaN      NaN      NaN      NaN   \n",
       "2        NaN  0.0002   0.0000      0.0      0.0      0.0    0.001   \n",
       "3        NaN  0.0002   0.0000      0.0      0.0      0.0    0.001   \n",
       "4       7/11  0.0002   0.0000      0.0      0.0      0.0    0.001   \n",
       "...      ...     ...      ...      ...      ...      ...      ...   \n",
       "10444   9/22  0.0042   0.0159      0.0      0.0      0.0    0.000   \n",
       "10445   9/22  0.0042   0.0159      0.0      0.0      0.0    0.000   \n",
       "10446   9/22  0.0042   0.0159      0.0      0.0      0.0    0.000   \n",
       "10447   9/22  0.0042   0.0159      0.0      0.0      0.0    0.000   \n",
       "10448   9/22  0.0042   0.0159      0.0      0.0      0.0    0.000   \n",
       "\n",
       "                           CLIN_SIG SOMATIC    PUBMED  HGVS_OFFSET PHENO  \\\n",
       "0                               NaN     NaN       NaN          NaN   NaN   \n",
       "1                               NaN     NaN       NaN          NaN   NaN   \n",
       "2                               NaN     NaN       NaN          NaN   NaN   \n",
       "3                               NaN     NaN       NaN          NaN   NaN   \n",
       "4                               NaN     NaN       NaN          NaN   NaN   \n",
       "...                             ...     ...       ...          ...   ...   \n",
       "10444  likely_pathogenic;pathogenic     1;0  25105660          1.0   1;0   \n",
       "10445  likely_pathogenic;pathogenic     1;0  25105660          1.0   1;0   \n",
       "10446  likely_pathogenic;pathogenic     1;0  25105660          1.0   1;0   \n",
       "10447  likely_pathogenic;pathogenic     1;0  25105660          1.0   1;0   \n",
       "10448  likely_pathogenic;pathogenic     1;0  25105660          1.0   1;0   \n",
       "\n",
       "                   GDC_FILTER  \n",
       "0                         NaN  \n",
       "1                         NaN  \n",
       "2              common_in_exac  \n",
       "3              common_in_exac  \n",
       "4                   NonExonic  \n",
       "...                       ...  \n",
       "10444  common_in_exac;gdc_pon  \n",
       "10445  common_in_exac;gdc_pon  \n",
       "10446  common_in_exac;gdc_pon  \n",
       "10447  common_in_exac;gdc_pon  \n",
       "10448  common_in_exac;gdc_pon  \n",
       "\n",
       "[10449 rows x 17 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Fill NaN values with interpolation method pad-by using existing values#\n",
    "X = X.interpolate(method=\"pad\")\n",
    "X.info()\n",
    "print(X.isna().any().sum())\n",
    "m = X.loc[:,X.isna().any()]\n",
    "m"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "29fe8677",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Columns: 119508 entries, Entrez_Gene_Id to GDC_Validation_Status_Valid\n",
      "dtypes: bool(1), float64(1), int64(10), uint8(119496)\n",
      "memory usage: 1.2 GB\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>TRANSCRIPT_STRAND</th>\n",
       "      <th>TSL</th>\n",
       "      <th>...</th>\n",
       "      <th>case_id_d9952e44-b573-4854-893e-b3bbc5287f2f</th>\n",
       "      <th>case_id_ddfdd05e-058c-4877-b389-724a6c101353</th>\n",
       "      <th>case_id_ebb834c6-374e-4077-a4ce-c74a850be358</th>\n",
       "      <th>case_id_ee6083aa-9453-44ff-8cc5-79b71fea289d</th>\n",
       "      <th>case_id_eec002c4-7602-4179-8eb3-2b0508d5ee1c</th>\n",
       "      <th>case_id_f2cc142b-9fc0-44d9-9ae3-558b4eace375</th>\n",
       "      <th>case_id_f98c183e-aaa1-40c4-83e7-c248cc3adb55</th>\n",
       "      <th>case_id_ffedc8be-1056-4205-b9d9-99b5bdb872db</th>\n",
       "      <th>GDC_Validation_Status_Unknown</th>\n",
       "      <th>GDC_Validation_Status_Valid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6536</td>\n",
       "      <td>44008472</td>\n",
       "      <td>44008472</td>\n",
       "      <td>103</td>\n",
       "      <td>65</td>\n",
       "      <td>38</td>\n",
       "      <td>132</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1263</td>\n",
       "      <td>44803280</td>\n",
       "      <td>44803280</td>\n",
       "      <td>84</td>\n",
       "      <td>51</td>\n",
       "      <td>33</td>\n",
       "      <td>112</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>199920</td>\n",
       "      <td>56740711</td>\n",
       "      <td>56740711</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2629</td>\n",
       "      <td>155233116</td>\n",
       "      <td>155233116</td>\n",
       "      <td>46</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>96</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>83417</td>\n",
       "      <td>157581530</td>\n",
       "      <td>157581530</td>\n",
       "      <td>60</td>\n",
       "      <td>33</td>\n",
       "      <td>27</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10444</th>\n",
       "      <td>6247</td>\n",
       "      <td>18642139</td>\n",
       "      <td>18642139</td>\n",
       "      <td>102</td>\n",
       "      <td>48</td>\n",
       "      <td>54</td>\n",
       "      <td>119</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10445</th>\n",
       "      <td>7317</td>\n",
       "      <td>47201524</td>\n",
       "      <td>47201524</td>\n",
       "      <td>43</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10446</th>\n",
       "      <td>441531</td>\n",
       "      <td>77969434</td>\n",
       "      <td>77969434</td>\n",
       "      <td>205</td>\n",
       "      <td>112</td>\n",
       "      <td>93</td>\n",
       "      <td>216</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10447</th>\n",
       "      <td>9737</td>\n",
       "      <td>102657224</td>\n",
       "      <td>102657224</td>\n",
       "      <td>199</td>\n",
       "      <td>97</td>\n",
       "      <td>102</td>\n",
       "      <td>186</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10448</th>\n",
       "      <td>114824</td>\n",
       "      <td>152991432</td>\n",
       "      <td>152991432</td>\n",
       "      <td>161</td>\n",
       "      <td>104</td>\n",
       "      <td>57</td>\n",
       "      <td>170</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10449 rows × 119508 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Entrez_Gene_Id  Start_Position  End_Position  t_depth  t_ref_count  \\\n",
       "0                6536        44008472      44008472      103           65   \n",
       "1                1263        44803280      44803280       84           51   \n",
       "2              199920        56740711      56740711       22           15   \n",
       "3                2629       155233116     155233116       46           29   \n",
       "4               83417       157581530     157581530       60           33   \n",
       "...               ...             ...           ...      ...          ...   \n",
       "10444            6247        18642139      18642139      102           48   \n",
       "10445            7317        47201524      47201524       43           32   \n",
       "10446          441531        77969434      77969434      205          112   \n",
       "10447            9737       102657224     102657224      199           97   \n",
       "10448          114824       152991432     152991432      161          104   \n",
       "\n",
       "       t_alt_count  n_depth  ALLELE_NUM  TRANSCRIPT_STRAND  TSL  ...  \\\n",
       "0               38      132           1                 -1  1.0  ...   \n",
       "1               33      112           1                  1  1.0  ...   \n",
       "2                7       23           1                 -1  1.0  ...   \n",
       "3               17       96           1                 -1  1.0  ...   \n",
       "4               27       79           1                 -1  1.0  ...   \n",
       "...            ...      ...         ...                ...  ...  ...   \n",
       "10444           54      119           1                 -1  1.0  ...   \n",
       "10445           11       47           1                  1  1.0  ...   \n",
       "10446           93      216           1                 -1  1.0  ...   \n",
       "10447          102      186           1                  1  2.0  ...   \n",
       "10448           57      170           1                 -1  1.0  ...   \n",
       "\n",
       "       case_id_d9952e44-b573-4854-893e-b3bbc5287f2f  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ddfdd05e-058c-4877-b389-724a6c101353  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ebb834c6-374e-4077-a4ce-c74a850be358  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ee6083aa-9453-44ff-8cc5-79b71fea289d  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_eec002c4-7602-4179-8eb3-2b0508d5ee1c  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_f2cc142b-9fc0-44d9-9ae3-558b4eace375  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_f98c183e-aaa1-40c4-83e7-c248cc3adb55  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ffedc8be-1056-4205-b9d9-99b5bdb872db  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       GDC_Validation_Status_Unknown  GDC_Validation_Status_Valid  \n",
       "0                                  0                            1  \n",
       "1                                  0                            1  \n",
       "2                                  1                            0  \n",
       "3                                  1                            0  \n",
       "4                                  1                            0  \n",
       "...                              ...                          ...  \n",
       "10444                              1                            0  \n",
       "10445                              1                            0  \n",
       "10446                              1                            0  \n",
       "10447                              1                            0  \n",
       "10448                              1                            0  \n",
       "\n",
       "[10449 rows x 119508 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obj_df = X.select_dtypes(include=['object'])\n",
    "X=pd.get_dummies(data=X,columns=obj_df.columns)\n",
    "X.info()\n",
    "X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "00d76037",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([5426, 4497,  734, ..., 4374, 2387, 4526])"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Y = LabelEncoder().fit_transform(Y.ravel())\n",
    "Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "bccb6e03",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-8-20a59aae736a>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mX\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mStandardScaler\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfit_transform\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mX\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\sklearn\\base.py\u001b[0m in \u001b[0;36mfit_transform\u001b[1;34m(self, X, y, **fit_params)\u001b[0m\n\u001b[0;32m    697\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0my\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    698\u001b[0m             \u001b[1;31m# fit method of arity 1 (unsupervised transformation)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 699\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mfit_params\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtransform\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    700\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    701\u001b[0m             \u001b[1;31m# fit method of arity 2 (supervised transformation)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\sklearn\\preprocessing\\_data.py\u001b[0m in \u001b[0;36mfit\u001b[1;34m(self, X, y, sample_weight)\u001b[0m\n\u001b[0;32m    728\u001b[0m         \u001b[1;31m# Reset internal state before fitting\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    729\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_reset\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 730\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpartial_fit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msample_weight\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    731\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    732\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mpartial_fit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mX\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0my\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0msample_weight\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\sklearn\\preprocessing\\_data.py\u001b[0m in \u001b[0;36mpartial_fit\u001b[1;34m(self, X, y, sample_weight)\u001b[0m\n\u001b[0;32m    764\u001b[0m         \"\"\"\n\u001b[0;32m    765\u001b[0m         \u001b[0mfirst_call\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mhasattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"n_samples_seen_\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 766\u001b[1;33m         X = self._validate_data(X, accept_sparse=('csr', 'csc'),\n\u001b[0m\u001b[0;32m    767\u001b[0m                                 \u001b[0mestimator\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mFLOAT_DTYPES\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    768\u001b[0m                                 force_all_finite='allow-nan', reset=first_call)\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\sklearn\\base.py\u001b[0m in \u001b[0;36m_validate_data\u001b[1;34m(self, X, y, reset, validate_separately, **check_params)\u001b[0m\n\u001b[0;32m    419\u001b[0m             \u001b[0mout\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mX\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    420\u001b[0m         \u001b[1;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0my\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0my\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'no_validation'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 421\u001b[1;33m             \u001b[0mX\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcheck_array\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mX\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mcheck_params\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    422\u001b[0m             \u001b[0mout\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mX\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    423\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\u001b[0m in \u001b[0;36minner_f\u001b[1;34m(*args, **kwargs)\u001b[0m\n\u001b[0;32m     61\u001b[0m             \u001b[0mextra_args\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m-\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mall_args\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     62\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mextra_args\u001b[0m \u001b[1;33m<=\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 63\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     64\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     65\u001b[0m             \u001b[1;31m# extra_args > 0\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\u001b[0m in \u001b[0;36mcheck_array\u001b[1;34m(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator)\u001b[0m\n\u001b[0;32m    671\u001b[0m                     \u001b[0marray\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0marray\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcasting\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"unsafe\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    672\u001b[0m                 \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 673\u001b[1;33m                     \u001b[0marray\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0marray\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morder\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0morder\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    674\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0mComplexWarning\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mcomplex_warning\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    675\u001b[0m                 raise ValueError(\"Complex data not supported\\n\"\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\numpy\\core\\_asarray.py\u001b[0m in \u001b[0;36masarray\u001b[1;34m(a, dtype, order, like)\u001b[0m\n\u001b[0;32m    100\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0m_asarray_with_like\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ma\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morder\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0morder\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlike\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mlike\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    101\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 102\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[0marray\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0ma\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0morder\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0morder\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    103\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    104\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m__array__\u001b[1;34m(self, dtype)\u001b[0m\n\u001b[0;32m   1988\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1989\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__array__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mNpDtype\u001b[0m \u001b[1;33m|\u001b[0m \u001b[1;32mNone\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1990\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1991\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1992\u001b[0m     def __array_wrap__(\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py\u001b[0m in \u001b[0;36m_values\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    892\u001b[0m         \u001b[0mblocks\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmgr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mblocks\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    893\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mblocks\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 894\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    895\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    896\u001b[0m         \u001b[0marr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mblocks\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\frame.py\u001b[0m in \u001b[0;36mvalues\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m  10659\u001b[0m         \"\"\"\n\u001b[0;32m  10660\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_consolidate_inplace\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m> 10661\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_mgr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mas_array\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtranspose\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m  10662\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m  10663\u001b[0m     \u001b[1;33m@\u001b[0m\u001b[0mdeprecate_nonkeyword_arguments\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mversion\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mallowed_args\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"self\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\internals\\managers.py\u001b[0m in \u001b[0;36mas_array\u001b[1;34m(self, transpose, dtype, copy, na_value)\u001b[0m\n\u001b[0;32m   1460\u001b[0m                     \u001b[0marr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0marr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mFalse\u001b[0m\u001b[1;33m)\u001b[0m  \u001b[1;31m# type: ignore[arg-type]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1461\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1462\u001b[1;33m             \u001b[0marr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_interleave\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mna_value\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mna_value\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1463\u001b[0m             \u001b[1;31m# The underlying data was copied within _interleave\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1464\u001b[0m             \u001b[0mcopy\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mFalse\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\internals\\managers.py\u001b[0m in \u001b[0;36m_interleave\u001b[1;34m(self, dtype, na_value)\u001b[0m\n\u001b[0;32m   1498\u001b[0m         \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mempty\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[1;33m)\u001b[0m  \u001b[1;31m# type: ignore[arg-type]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1499\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1500\u001b[1;33m         \u001b[0mitemmask\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mzeros\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1501\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1502\u001b[0m         \u001b[1;32mfor\u001b[0m \u001b[0mblk\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mblocks\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "X = StandardScaler().fit_transform(X)\n",
    "X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97a64e7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def forest_test(X, Y):\n",
    "    X_Train, X_Test, Y_Train, Y_Test = train_test_split(X, Y, \n",
    "                                                        test_size = 0.20, \n",
    "                                                        random_state = 101)\n",
    "    start = time.process_time()\n",
    "    trainedforest = RandomForestClassifier(n_estimators=100).fit(X_Train,Y_Train)\n",
    "    print(time.process_time() - start)\n",
    "    predictionforest = trainedforest.predict(X_Test)\n",
    "    print(confusion_matrix(Y_Test,predictionforest))\n",
    "    print(classification_report(Y_Test,predictionforest))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "188b2606",
   "metadata": {},
   "outputs": [],
   "source": [
    "forest_test(X, Y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc39d6bd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
