{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6f74605b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<gzip _io.BufferedReader name='SNV/02747363-f04a-4ba6-a079-fe4f87853788/TCGA.UCS.mutect.02747363-f04a-4ba6-a079-fe4f87853788.DR-10.0.somatic.maf.gz' 0x1f485db5bb0>\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import time\n",
    "import pandas as pd\n",
    "from sklearn import preprocessing\n",
    "from sklearn.impute import SimpleImputer\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.preprocessing import Normalizer\n",
    "from sklearn.preprocessing import OneHotEncoder\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "import matplotlib.pyplot as plt\n",
    "import gzip\n",
    "\n",
    "df = pd.read_table(\"SNV/MANIFEST.txt\")\n",
    "fileslist = []\n",
    "header = []\n",
    "for file_name in df[0:33][\"filename\"]:\n",
    "    fileslist.append(file_name)\n",
    "with gzip.open(\"SNV/\"+fileslist[0],\"rb\") as f:\n",
    "    print(f)\n",
    "    i = 0\n",
    "    while i <=4:\n",
    "        header.append(f.readline())\n",
    "        i += 1\n",
    "    mf = pd.read_csv(f, delimiter = \"\\t\")\n",
    "    table = mf[0:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cb5a8c33",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "54\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Data columns (total 102 columns):\n",
      " #    Column                       Dtype  \n",
      "---   ------                       -----  \n",
      " 0    Hugo_Symbol                  object \n",
      " 1    Entrez_Gene_Id               int64  \n",
      " 2    Center                       object \n",
      " 3    NCBI_Build                   object \n",
      " 4    Chromosome                   object \n",
      " 5    Start_Position               int64  \n",
      " 6    End_Position                 int64  \n",
      " 7    Strand                       object \n",
      " 8    Variant_Classification       object \n",
      " 9    Variant_Type                 object \n",
      " 10   Reference_Allele             object \n",
      " 11   Tumor_Seq_Allele1            object \n",
      " 12   Tumor_Seq_Allele2            object \n",
      " 13   dbSNP_RS                     object \n",
      " 14   dbSNP_Val_Status             object \n",
      " 15   Tumor_Sample_Barcode         object \n",
      " 16   Matched_Norm_Sample_Barcode  object \n",
      " 17   Tumor_Validation_Allele1     object \n",
      " 18   Tumor_Validation_Allele2     object \n",
      " 19   Mutation_Status              object \n",
      " 20   Validation_Method            object \n",
      " 21   Sequencer                    object \n",
      " 22   Tumor_Sample_UUID            object \n",
      " 23   Matched_Norm_Sample_UUID     object \n",
      " 24   HGVSc                        object \n",
      " 25   HGVSp                        object \n",
      " 26   HGVSp_Short                  object \n",
      " 27   Transcript_ID                object \n",
      " 28   Exon_Number                  object \n",
      " 29   t_depth                      int64  \n",
      " 30   t_ref_count                  int64  \n",
      " 31   t_alt_count                  int64  \n",
      " 32   n_depth                      int64  \n",
      " 33   all_effects                  object \n",
      " 34   Allele                       object \n",
      " 35   Gene                         object \n",
      " 36   Feature                      object \n",
      " 37   Feature_type                 object \n",
      " 38   One_Consequence              object \n",
      " 39   Consequence                  object \n",
      " 40   cDNA_position                object \n",
      " 41   CDS_position                 object \n",
      " 42   Protein_position             object \n",
      " 43   Amino_acids                  object \n",
      " 44   Codons                       object \n",
      " 45   Existing_variation           object \n",
      " 46   ALLELE_NUM                   int64  \n",
      " 47   DISTANCE                     float64\n",
      " 48   TRANSCRIPT_STRAND            int64  \n",
      " 49   SYMBOL                       object \n",
      " 50   SYMBOL_SOURCE                object \n",
      " 51   HGNC_ID                      object \n",
      " 52   BIOTYPE                      object \n",
      " 53   CANONICAL                    object \n",
      " 54   CCDS                         object \n",
      " 55   ENSP                         object \n",
      " 56   SWISSPROT                    object \n",
      " 57   TREMBL                       object \n",
      " 58   UNIPARC                      object \n",
      " 59   RefSeq                       object \n",
      " 60   SIFT                         object \n",
      " 61   PolyPhen                     object \n",
      " 62   EXON                         object \n",
      " 63   INTRON                       object \n",
      " 64   DOMAINS                      object \n",
      " 65   GMAF                         float64\n",
      " 66   AFR_MAF                      float64\n",
      " 67   AMR_MAF                      float64\n",
      " 68   EAS_MAF                      float64\n",
      " 69   EUR_MAF                      float64\n",
      " 70   SAS_MAF                      float64\n",
      " 71   AA_MAF                       float64\n",
      " 72   EA_MAF                       float64\n",
      " 73   CLIN_SIG                     object \n",
      " 74   SOMATIC                      object \n",
      " 75   PUBMED                       object \n",
      " 76   IMPACT                       object \n",
      " 77   PICK                         float64\n",
      " 78   VARIANT_CLASS                object \n",
      " 79   TSL                          float64\n",
      " 80   HGVS_OFFSET                  float64\n",
      " 81   PHENO                        object \n",
      " 82   MINIMISED                    int64  \n",
      " 83   ExAC_AF                      float64\n",
      " 84   ExAC_AF_Adj                  float64\n",
      " 85   ExAC_AF_AFR                  float64\n",
      " 86   ExAC_AF_AMR                  float64\n",
      " 87   ExAC_AF_EAS                  float64\n",
      " 88   ExAC_AF_FIN                  float64\n",
      " 89   ExAC_AF_NFE                  float64\n",
      " 90   ExAC_AF_OTH                  float64\n",
      " 91   ExAC_AF_SAS                  float64\n",
      " 92   FILTER                       object \n",
      " 93   CONTEXT                      object \n",
      " 94   src_vcf_id                   object \n",
      " 95   tumor_bam_uuid               object \n",
      " 96   normal_bam_uuid              object \n",
      " 97   case_id                      object \n",
      " 98   GDC_FILTER                   object \n",
      " 99   COSMIC                       object \n",
      " 100  MC3_Overlap                  bool   \n",
      " 101  GDC_Validation_Status        object \n",
      "dtypes: bool(1), float64(21), int64(10), object(70)\n",
      "memory usage: 8.1+ MB\n"
     ]
    }
   ],
   "source": [
    "#Remove columns with all row values NaN\n",
    "new_table = table.dropna(axis=1,how=\"all\") \n",
    "print(new_table.isna().any().sum())\n",
    "new_table.info(verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a6f4b3c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Data columns (total 101 columns):\n",
      " #    Column                       Dtype  \n",
      "---   ------                       -----  \n",
      " 0    Entrez_Gene_Id               int64  \n",
      " 1    Center                       object \n",
      " 2    NCBI_Build                   object \n",
      " 3    Chromosome                   object \n",
      " 4    Start_Position               int64  \n",
      " 5    End_Position                 int64  \n",
      " 6    Strand                       object \n",
      " 7    Variant_Classification       object \n",
      " 8    Variant_Type                 object \n",
      " 9    Reference_Allele             object \n",
      " 10   Tumor_Seq_Allele1            object \n",
      " 11   Tumor_Seq_Allele2            object \n",
      " 12   dbSNP_RS                     object \n",
      " 13   dbSNP_Val_Status             object \n",
      " 14   Tumor_Sample_Barcode         object \n",
      " 15   Matched_Norm_Sample_Barcode  object \n",
      " 16   Tumor_Validation_Allele1     object \n",
      " 17   Tumor_Validation_Allele2     object \n",
      " 18   Mutation_Status              object \n",
      " 19   Validation_Method            object \n",
      " 20   Sequencer                    object \n",
      " 21   Tumor_Sample_UUID            object \n",
      " 22   Matched_Norm_Sample_UUID     object \n",
      " 23   HGVSc                        object \n",
      " 24   HGVSp                        object \n",
      " 25   HGVSp_Short                  object \n",
      " 26   Transcript_ID                object \n",
      " 27   Exon_Number                  object \n",
      " 28   t_depth                      int64  \n",
      " 29   t_ref_count                  int64  \n",
      " 30   t_alt_count                  int64  \n",
      " 31   n_depth                      int64  \n",
      " 32   all_effects                  object \n",
      " 33   Allele                       object \n",
      " 34   Gene                         object \n",
      " 35   Feature                      object \n",
      " 36   Feature_type                 object \n",
      " 37   One_Consequence              object \n",
      " 38   Consequence                  object \n",
      " 39   cDNA_position                object \n",
      " 40   CDS_position                 object \n",
      " 41   Protein_position             object \n",
      " 42   Amino_acids                  object \n",
      " 43   Codons                       object \n",
      " 44   Existing_variation           object \n",
      " 45   ALLELE_NUM                   int64  \n",
      " 46   DISTANCE                     float64\n",
      " 47   TRANSCRIPT_STRAND            int64  \n",
      " 48   SYMBOL                       object \n",
      " 49   SYMBOL_SOURCE                object \n",
      " 50   HGNC_ID                      object \n",
      " 51   BIOTYPE                      object \n",
      " 52   CANONICAL                    object \n",
      " 53   CCDS                         object \n",
      " 54   ENSP                         object \n",
      " 55   SWISSPROT                    object \n",
      " 56   TREMBL                       object \n",
      " 57   UNIPARC                      object \n",
      " 58   RefSeq                       object \n",
      " 59   SIFT                         object \n",
      " 60   PolyPhen                     object \n",
      " 61   EXON                         object \n",
      " 62   INTRON                       object \n",
      " 63   DOMAINS                      object \n",
      " 64   GMAF                         float64\n",
      " 65   AFR_MAF                      float64\n",
      " 66   AMR_MAF                      float64\n",
      " 67   EAS_MAF                      float64\n",
      " 68   EUR_MAF                      float64\n",
      " 69   SAS_MAF                      float64\n",
      " 70   AA_MAF                       float64\n",
      " 71   EA_MAF                       float64\n",
      " 72   CLIN_SIG                     object \n",
      " 73   SOMATIC                      object \n",
      " 74   PUBMED                       object \n",
      " 75   IMPACT                       object \n",
      " 76   PICK                         float64\n",
      " 77   VARIANT_CLASS                object \n",
      " 78   TSL                          float64\n",
      " 79   HGVS_OFFSET                  float64\n",
      " 80   PHENO                        object \n",
      " 81   MINIMISED                    int64  \n",
      " 82   ExAC_AF                      float64\n",
      " 83   ExAC_AF_Adj                  float64\n",
      " 84   ExAC_AF_AFR                  float64\n",
      " 85   ExAC_AF_AMR                  float64\n",
      " 86   ExAC_AF_EAS                  float64\n",
      " 87   ExAC_AF_FIN                  float64\n",
      " 88   ExAC_AF_NFE                  float64\n",
      " 89   ExAC_AF_OTH                  float64\n",
      " 90   ExAC_AF_SAS                  float64\n",
      " 91   FILTER                       object \n",
      " 92   CONTEXT                      object \n",
      " 93   src_vcf_id                   object \n",
      " 94   tumor_bam_uuid               object \n",
      " 95   normal_bam_uuid              object \n",
      " 96   case_id                      object \n",
      " 97   GDC_FILTER                   object \n",
      " 98   COSMIC                       object \n",
      " 99   MC3_Overlap                  bool   \n",
      " 100  GDC_Validation_Status        object \n",
      "dtypes: bool(1), float64(21), int64(10), object(69)\n",
      "memory usage: 8.0+ MB\n"
     ]
    }
   ],
   "source": [
    "#Handle features and genes_id#\n",
    "\n",
    "#Retrieve Y (genes_id) and X (all other features)\n",
    "Y = new_table.loc[:,[\"Hugo_Symbol\"]].values\n",
    "length = len(Y)\n",
    "\n",
    "#Retrive features\n",
    "X = new_table.drop([\"Hugo_Symbol\"], axis = 1)\n",
    "thresh = length*0.90\n",
    "X.info(verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6f076c7e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Data columns (total 58 columns):\n",
      " #   Column                       Non-Null Count  Dtype  \n",
      "---  ------                       --------------  -----  \n",
      " 0   Entrez_Gene_Id               10449 non-null  int64  \n",
      " 1   Center                       10449 non-null  object \n",
      " 2   NCBI_Build                   10449 non-null  object \n",
      " 3   Chromosome                   10449 non-null  object \n",
      " 4   Start_Position               10449 non-null  int64  \n",
      " 5   End_Position                 10449 non-null  int64  \n",
      " 6   Strand                       10449 non-null  object \n",
      " 7   Variant_Classification       10449 non-null  object \n",
      " 8   Variant_Type                 10449 non-null  object \n",
      " 9   Reference_Allele             10449 non-null  object \n",
      " 10  Tumor_Seq_Allele1            10449 non-null  object \n",
      " 11  Tumor_Seq_Allele2            10449 non-null  object \n",
      " 12  dbSNP_RS                     9507 non-null   object \n",
      " 13  Tumor_Sample_Barcode         10449 non-null  object \n",
      " 14  Matched_Norm_Sample_Barcode  10449 non-null  object \n",
      " 15  Mutation_Status              10449 non-null  object \n",
      " 16  Sequencer                    10449 non-null  object \n",
      " 17  Tumor_Sample_UUID            10449 non-null  object \n",
      " 18  Matched_Norm_Sample_UUID     10449 non-null  object \n",
      " 19  HGVSc                        10364 non-null  object \n",
      " 20  Transcript_ID                10449 non-null  object \n",
      " 21  Exon_Number                  9828 non-null   object \n",
      " 22  t_depth                      10449 non-null  int64  \n",
      " 23  t_ref_count                  10449 non-null  int64  \n",
      " 24  t_alt_count                  10449 non-null  int64  \n",
      " 25  n_depth                      10449 non-null  int64  \n",
      " 26  all_effects                  10449 non-null  object \n",
      " 27  Allele                       10449 non-null  object \n",
      " 28  Gene                         10449 non-null  object \n",
      " 29  Feature                      10449 non-null  object \n",
      " 30  Feature_type                 10449 non-null  object \n",
      " 31  One_Consequence              10449 non-null  object \n",
      " 32  Consequence                  10449 non-null  object \n",
      " 33  cDNA_position                10449 non-null  object \n",
      " 34  CDS_position                 10247 non-null  object \n",
      " 35  Protein_position             10247 non-null  object \n",
      " 36  ALLELE_NUM                   10449 non-null  int64  \n",
      " 37  TRANSCRIPT_STRAND            10449 non-null  int64  \n",
      " 38  SYMBOL                       10449 non-null  object \n",
      " 39  SYMBOL_SOURCE                10449 non-null  object \n",
      " 40  HGNC_ID                      10369 non-null  object \n",
      " 41  BIOTYPE                      10449 non-null  object \n",
      " 42  CCDS                         9760 non-null   object \n",
      " 43  ENSP                         10247 non-null  object \n",
      " 44  SWISSPROT                    9934 non-null   object \n",
      " 45  EXON                         9828 non-null   object \n",
      " 46  IMPACT                       10449 non-null  object \n",
      " 47  VARIANT_CLASS                10449 non-null  object \n",
      " 48  TSL                          9823 non-null   float64\n",
      " 49  MINIMISED                    10449 non-null  int64  \n",
      " 50  FILTER                       10449 non-null  object \n",
      " 51  CONTEXT                      10449 non-null  object \n",
      " 52  src_vcf_id                   10449 non-null  object \n",
      " 53  tumor_bam_uuid               10449 non-null  object \n",
      " 54  normal_bam_uuid              10449 non-null  object \n",
      " 55  case_id                      10449 non-null  object \n",
      " 56  MC3_Overlap                  10449 non-null  bool   \n",
      " 57  GDC_Validation_Status        10449 non-null  object \n",
      "dtypes: bool(1), float64(1), int64(10), object(46)\n",
      "memory usage: 4.6+ MB\n"
     ]
    }
   ],
   "source": [
    "#Handling NaN in cloumns below treshold 0.90(90%)#\n",
    "X  = X.dropna(axis=1,thresh=int(thresh))\n",
    "#X.isna().sum().sort_values(ascending=False)\n",
    "X.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c1d9093c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Columns: 119508 entries, Entrez_Gene_Id to GDC_Validation_Status_Valid\n",
      "dtypes: bool(1), float64(1), int64(10), uint8(119496)\n",
      "memory usage: 1.2 GB\n"
     ]
    }
   ],
   "source": [
    "#Fill NaN values with interpolation method pad-by using existing values#\n",
    "X = X.interpolate(method=\"pad\")\n",
    "X.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "29fe8677",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10449 entries, 0 to 10448\n",
      "Columns: 119508 entries, Entrez_Gene_Id to GDC_Validation_Status_Valid\n",
      "dtypes: bool(1), float64(1), int64(10), uint8(119496)\n",
      "memory usage: 1.2 GB\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>TRANSCRIPT_STRAND</th>\n",
       "      <th>TSL</th>\n",
       "      <th>...</th>\n",
       "      <th>case_id_d9952e44-b573-4854-893e-b3bbc5287f2f</th>\n",
       "      <th>case_id_ddfdd05e-058c-4877-b389-724a6c101353</th>\n",
       "      <th>case_id_ebb834c6-374e-4077-a4ce-c74a850be358</th>\n",
       "      <th>case_id_ee6083aa-9453-44ff-8cc5-79b71fea289d</th>\n",
       "      <th>case_id_eec002c4-7602-4179-8eb3-2b0508d5ee1c</th>\n",
       "      <th>case_id_f2cc142b-9fc0-44d9-9ae3-558b4eace375</th>\n",
       "      <th>case_id_f98c183e-aaa1-40c4-83e7-c248cc3adb55</th>\n",
       "      <th>case_id_ffedc8be-1056-4205-b9d9-99b5bdb872db</th>\n",
       "      <th>GDC_Validation_Status_Unknown</th>\n",
       "      <th>GDC_Validation_Status_Valid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6536</td>\n",
       "      <td>44008472</td>\n",
       "      <td>44008472</td>\n",
       "      <td>103</td>\n",
       "      <td>65</td>\n",
       "      <td>38</td>\n",
       "      <td>132</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1263</td>\n",
       "      <td>44803280</td>\n",
       "      <td>44803280</td>\n",
       "      <td>84</td>\n",
       "      <td>51</td>\n",
       "      <td>33</td>\n",
       "      <td>112</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>199920</td>\n",
       "      <td>56740711</td>\n",
       "      <td>56740711</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2629</td>\n",
       "      <td>155233116</td>\n",
       "      <td>155233116</td>\n",
       "      <td>46</td>\n",
       "      <td>29</td>\n",
       "      <td>17</td>\n",
       "      <td>96</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>83417</td>\n",
       "      <td>157581530</td>\n",
       "      <td>157581530</td>\n",
       "      <td>60</td>\n",
       "      <td>33</td>\n",
       "      <td>27</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10444</th>\n",
       "      <td>6247</td>\n",
       "      <td>18642139</td>\n",
       "      <td>18642139</td>\n",
       "      <td>102</td>\n",
       "      <td>48</td>\n",
       "      <td>54</td>\n",
       "      <td>119</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10445</th>\n",
       "      <td>7317</td>\n",
       "      <td>47201524</td>\n",
       "      <td>47201524</td>\n",
       "      <td>43</td>\n",
       "      <td>32</td>\n",
       "      <td>11</td>\n",
       "      <td>47</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10446</th>\n",
       "      <td>441531</td>\n",
       "      <td>77969434</td>\n",
       "      <td>77969434</td>\n",
       "      <td>205</td>\n",
       "      <td>112</td>\n",
       "      <td>93</td>\n",
       "      <td>216</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10447</th>\n",
       "      <td>9737</td>\n",
       "      <td>102657224</td>\n",
       "      <td>102657224</td>\n",
       "      <td>199</td>\n",
       "      <td>97</td>\n",
       "      <td>102</td>\n",
       "      <td>186</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10448</th>\n",
       "      <td>114824</td>\n",
       "      <td>152991432</td>\n",
       "      <td>152991432</td>\n",
       "      <td>161</td>\n",
       "      <td>104</td>\n",
       "      <td>57</td>\n",
       "      <td>170</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10449 rows Ã— 119508 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Entrez_Gene_Id  Start_Position  End_Position  t_depth  t_ref_count  \\\n",
       "0                6536        44008472      44008472      103           65   \n",
       "1                1263        44803280      44803280       84           51   \n",
       "2              199920        56740711      56740711       22           15   \n",
       "3                2629       155233116     155233116       46           29   \n",
       "4               83417       157581530     157581530       60           33   \n",
       "...               ...             ...           ...      ...          ...   \n",
       "10444            6247        18642139      18642139      102           48   \n",
       "10445            7317        47201524      47201524       43           32   \n",
       "10446          441531        77969434      77969434      205          112   \n",
       "10447            9737       102657224     102657224      199           97   \n",
       "10448          114824       152991432     152991432      161          104   \n",
       "\n",
       "       t_alt_count  n_depth  ALLELE_NUM  TRANSCRIPT_STRAND  TSL  ...  \\\n",
       "0               38      132           1                 -1  1.0  ...   \n",
       "1               33      112           1                  1  1.0  ...   \n",
       "2                7       23           1                 -1  1.0  ...   \n",
       "3               17       96           1                 -1  1.0  ...   \n",
       "4               27       79           1                 -1  1.0  ...   \n",
       "...            ...      ...         ...                ...  ...  ...   \n",
       "10444           54      119           1                 -1  1.0  ...   \n",
       "10445           11       47           1                  1  1.0  ...   \n",
       "10446           93      216           1                 -1  1.0  ...   \n",
       "10447          102      186           1                  1  2.0  ...   \n",
       "10448           57      170           1                 -1  1.0  ...   \n",
       "\n",
       "       case_id_d9952e44-b573-4854-893e-b3bbc5287f2f  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ddfdd05e-058c-4877-b389-724a6c101353  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ebb834c6-374e-4077-a4ce-c74a850be358  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ee6083aa-9453-44ff-8cc5-79b71fea289d  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_eec002c4-7602-4179-8eb3-2b0508d5ee1c  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_f2cc142b-9fc0-44d9-9ae3-558b4eace375  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_f98c183e-aaa1-40c4-83e7-c248cc3adb55  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       case_id_ffedc8be-1056-4205-b9d9-99b5bdb872db  \\\n",
       "0                                                 0   \n",
       "1                                                 0   \n",
       "2                                                 0   \n",
       "3                                                 0   \n",
       "4                                                 0   \n",
       "...                                             ...   \n",
       "10444                                             0   \n",
       "10445                                             0   \n",
       "10446                                             0   \n",
       "10447                                             0   \n",
       "10448                                             0   \n",
       "\n",
       "       GDC_Validation_Status_Unknown  GDC_Validation_Status_Valid  \n",
       "0                                  0                            1  \n",
       "1                                  0                            1  \n",
       "2                                  1                            0  \n",
       "3                                  1                            0  \n",
       "4                                  1                            0  \n",
       "...                              ...                          ...  \n",
       "10444                              1                            0  \n",
       "10445                              1                            0  \n",
       "10446                              1                            0  \n",
       "10447                              1                            0  \n",
       "10448                              1                            0  \n",
       "\n",
       "[10449 rows x 119508 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obj_df = X.select_dtypes(include=['object'])\n",
    "X=pd.get_dummies(data=X,columns=obj_df.columns)\n",
    "X.info()\n",
    "X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "00d76037",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([5426, 4497,  734, ..., 4374, 2387, 4526])"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Y = LabelEncoder().fit_transform(Y.ravel())\n",
    "Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "bccb6e03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[-0.08414803, -0.61689314, -0.61689315, ..., -0.07599565,\n",
       "        -1.72511663,  1.72511663],\n",
       "       [-0.08483798, -0.60296662, -0.60296662, ..., -0.07599565,\n",
       "        -1.72511663,  1.72511663],\n",
       "       [-0.0588445 , -0.39380041, -0.39380041, ..., -0.07599565,\n",
       "         0.57967095, -0.57967095],\n",
       "       ...,\n",
       "       [-0.02723065, -0.02183333, -0.02183333, ..., -0.07599565,\n",
       "         0.57967095, -0.57967095],\n",
       "       [-0.08372919,  0.4107431 ,  0.4107431 , ..., -0.07599565,\n",
       "         0.57967095, -0.57967095],\n",
       "       [-0.06997897,  1.29269292,  1.29269292, ..., -0.07599565,\n",
       "         0.57967095, -0.57967095]])"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X = StandardScaler().fit_transform(X)\n",
    "X"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97a64e7c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
